.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,26 @@
+name: Differential ShellCheck
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    permissions:
+      security-events: write
+      pull-requests: write
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Differential ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v3
```

Comment:
For other third-party code that gets a token we usually pin down an exact SHA instead of a (fungible) tag. However, token here does not have privileges which would allow secret infiltration, in the worst case it can cause spamming. So I think it's okay here. @allisonkarlitskaya do you agree?

Reply:
Pinning to SHA is good practice, but it is usually complemented by some tool (e.g. Dependabot) that looks for updates and notifies you when a new version of GitHub Action has been released.

Reply:
OK, let's keep it at `@v3` for now. Please also revert the SHA pinning of `actions/checkout@v2` -- that one is from GitHub itself. If they want to 0wn us, they have more discreet ways to do that :grin: